LEAF_Request_Portal/sources/Form.php

```diff
@@ -784,21 +804,28 @@ public function cancelRecord(int $recordID, ?string $comment = ''): int|string
         $return_value = 'Please contact your administrator';
 
         $vars = array(':recordID' => $recordID);
-        $sql = 'SELECT `submitted`
+        $sql = 'SELECT `userID`, `submitted`, `stepID`
                     FROM `records`
+                    LEFT JOIN records_workflow_state USING (`recordID`)
                     WHERE `recordID` = :recordID';
-        $resIsSubmitted = $this->db->prepared_query($sql, $vars);
+        $res = $this->db->prepared_query($sql, $vars);
+
+        if ($res[0]['submitted'] > 0 && $res[0]['stepID'] == null && !$this->login->checkGroup(1)) {
```

Comment:
Two concerns with this if operation.  First, the "else if" does a hard return.  This should just be putting a value to $return_value; each function should only have a single "return()".  Second, both conditionals are pretty complex.  I'm not seeing a final "else" for a catch-all.  I see that $return_value is declared above, but it might be a good idea to include the final "else" for clarity's sake and just in case.

Reply:
I understand the theory of single-return, I'm personally not a fan of always using it. In this situation, using a single return would make the actual logic changes harder to review.

For example, note that all the "changes" due to indentations make it harder to visualize here: https://github.com/department-of-veterans-affairs/LEAF/pull/2481/files